commands/sync.go

```diff
@@ -78,28 +79,50 @@ var SyncStatusCmd = &cli.Command{
 	},
 }
 
+type syncWaitOptions struct{
+	watch bool
+	untilEpoch int
+}
+
+var opts syncWaitOptions
+
 var SyncWaitCmd = &cli.Command{
 	Name:  "wait",
 	Usage: "Wait for sync to be complete",
 	Flags: []cli.Flag{
 		&cli.BoolFlag{
 			Name:  "watch",
 			Usage: "don't exit after node is synced",
+			Destination: &opts.watch,
+		},
+		&cli.IntFlag{
+			Name:  "until-epoch",
+			Usage: "returns after the epoch provided",
+			Destination: &opts.untilEpoch,
 		},
 	},
 	Action: func(cctx *cli.Context) error {
+		if opts.untilEpoch > 0 && opts.watch {
+			return fmt.Errorf("until-epoch and watch flags cannot be used together")
+		}
+		// ensure default doesn't cause an internal exit by making the value
+		// impossibly large
+		if opts.untilEpoch == 0 {
+			opts.untilEpoch = math.MaxInt64
+		}
```

Comment:
I didn't want to set MaxInt64 as the default in the arg because it would be misleading. This is janky, but as a quick hack it seems acceptable.